clients/src/main/java/org/apache/kafka/common/requests/FetchResponse.java

```diff
@@ -304,58 +115,12 @@ public int sessionId() {
     public Map<Errors, Integer> errorCounts() {
         Map<Errors, Integer> errorCounts = new HashMap<>();
         updateErrorCounts(errorCounts, error());
-        responseDataMap.values().forEach(response ->
-            updateErrorCounts(errorCounts, response.error())
-        );
+        dataByTopicPartition.values().forEach(response -> updateErrorCounts(errorCounts, Errors.forCode(response.errorCode())));
         return errorCounts;
     }
 
-    public static FetchResponse<MemoryRecords> parse(ByteBuffer buffer, short version) {
-        return new FetchResponse<>(new FetchResponseData(new ByteBufferAccessor(buffer), version));
-    }
-
-    @SuppressWarnings("unchecked")
-    private static <T extends BaseRecords> LinkedHashMap<TopicPartition, PartitionData<T>> toResponseDataMap(
-            FetchResponseData message) {
-        LinkedHashMap<TopicPartition, PartitionData<T>> responseMap = new LinkedHashMap<>();
-        message.responses().forEach(topicResponse -> {
-            topicResponse.partitionResponses().forEach(partitionResponse -> {
-                TopicPartition tp = new TopicPartition(topicResponse.topic(), partitionResponse.partition());
-                PartitionData<T> partitionData = new PartitionData<>(partitionResponse);
-                responseMap.put(tp, partitionData);
-            });
-        });
-        return responseMap;
-    }
-
-    private static <T extends BaseRecords> FetchResponseData toMessage(int throttleTimeMs, Errors error,
```

Comment:
I don't batch the partitions again in this PR as it create a new ```FetchResponseData```.

Reply:
Oh, I planed to remove all usages of this method from production (i.e `KafkaApis` should generate batched response directly). However, it can produce a big patch so I will keep this method in next commit.